iceoryx_posh/include/iceoryx_posh/internal/capro/capro_message.inl

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+namespace iox
+{
+namespace capro
+{
+inline constexpr const char* caproMessageTypeString(CaproMessageType messageType) noexcept
+{
+    switch (messageType)
+    {
+    case CaproMessageType::NOTYPE:
+        return "NOTYPE";
+    case CaproMessageType::FIND:
+        return "FIND";
+    case CaproMessageType::OFFER:
+        return "OFFER";
+    case CaproMessageType::STOP_OFFER:
+        return "STOP_OFFER";
+    case CaproMessageType::SUB:
+        return "SUB";
+    case CaproMessageType::UNSUB:
+        return "UNSUB";
+    case CaproMessageType::CONNECT:
+        return "CONNECT";
+    case CaproMessageType::DISCONNECT:
+        return "DISCONNECT";
+    case CaproMessageType::HANDSHAKE:
+        return "HANDSHAKE";
+    case CaproMessageType::ACK:
+        return "ACK";
+    case CaproMessageType::NACK:
+        return "NACK";
+    case CaproMessageType::PUB:
+        return "PUB";
+    case CaproMessageType::REQ:
+        return "REQ";
+    case CaproMessageType::RES:
+        return "RES";
+    case CaproMessageType::PING:
+        return "PING";
+    case CaproMessageType::PONG:
+        return "PONG";
+    case CaproMessageType::MESSGAGE_TYPE_END:
+        return "MESSGAGE_TYPE_END";
+    }
+
+    return "UNKNOWN_TYPE";
```

Comment:
Moving this into default would mean we might miss a new message added to the enum. Without a `default` case, the compiler will complain. There is still the return after the switch, in order to catch errors when a dummy enum value was provided. In this case, it's clear that MISRA is for C type enums and not `enum class` from C++

Reply:
Ok, can you please double check the Axivion run?

Reply:
ahem, let's assume I'm a new contributor ... how would I do that?

Reply:
This is describe in the `CONTRIBUTING.md`: Friendly ping one of the maintainers to get access to the [dashboard](https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#safety--security) ;) 

@dkroenke will probably give everyone some heads-up before this goes live with every PR in a couple of weeks. I'd appreciate if you give it a try for this PR :)

Reply:
You can find the config files with all the rules in [here](https://github.com/eclipse-iceoryx/iceoryx/blob/master/tools/axivion/axivion_config.json#L64)

Reply:
@mossmaurice according to dkroenke the dashboard currently only works with the master branch. We currently use this pattern (no default but return statement) in several places already, e.g. https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/source/popo/trigger.cpp#L32-L41.
Furthermore it is semantically the same as with the `default` case when all `enum` tags are handled and compiles down to the exact same [assembly](https://godbolt.org/z/zrf9snnfY) with the benefit of a compile time error instead of a silent fallback to `default` when the `enum` is extended.

Assuming our current rule set doesn't allow this, we need to change it anyway since it doesn't make sense and results in bad code where errors need to be catched by tests instead of the compiler.